src/main/scala/scodec/codecs/TupleCodec.scala

```diff
@@ -3,14 +3,401 @@ package codecs
 
 import scodec.bits.BitVector
 
-private[scodec] final class TupleCodec[A, B](codecA: Codec[A], codecB: Codec[B]) extends Codec[(A, B)] {
+private[scodec] final class TupleCodec[A, B](A: Codec[A], B: Codec[B]) extends Codec[(A, B)] {
```

Comment:
I think we need to remove the `private[scodec]` designations too -- otherwise, callers won't be able to see these types, and hence, no `~~` calls.